### PR TITLE
Fix #434 - No error when using `UPDATE ... SET`

### LINKED ISSUE #434: No error when using UPDATE ... SET
`UPDATE test SET WHERE` doesn't show any error.

### PR DESCRIPTION
**Pull request by niconoe-**
Fixes #434 

@williamdes / @iifawzi / @MauricioFauth / @liviuconcioiu 

This should also be fixed in 5.10.x and master. How do you want me to proceed to perform this?

### CODE REVIEW

**Comment by MauricioFauth** [comment]
Could you please target 5.10.x instead of 5.9.x? Thanks!

**Comment by niconoe-** [comment]
> Could you please target 5.10.x instead of 5.9.x? Thanks!

Done, but the bug will remain en 5.9.x after the merge…

Or do you plan to release a 5.10.0 soon ? :innocent: :yum:

**Review by williamdes — APPROVED** [review]

**Comment by niconoe-** [comment]
I updated the fix because it caused failures in SET operations out of an update statement context (like `SET NAMES …`).

Now, that's ok :smiley: . (except a linter in PHP 7.4 which is out of the scope of this PR :shrug: ).

File: src/Statements/UpdateStatement.php [hunk 1]
```diff
@@ -135,4 +139,23 @@ class UpdateStatement extends Statement
      * @var JoinKeyword[]|null
      */
     public $join;
+
+    /**
+     * Function called after the token was processed.
+     * In the update statement, this is used to check that at least one assignment has been set to throw an error if a
+     * query like `UPDATE acme SET WHERE 1;` is parsed.
+     *
+     * @throws ParserException
```
Comment (williamdes) [inline src/Statements/UpdateStatement.php:148]: really ?

Comment (niconoe-) [reply src/Statements/UpdateStatement.php:148]: It actually can throws the ParserException if strict-mode is enabled. This is due to `$parser->error` doing it exactly this way.

I'll add this information in order to make it clearer.

**Review by MauricioFauth — APPROVED** [review]

**Pull request merged**